### DISCUSSION TITLE
@midwayjs/passport 模块 isAuthenticated() 函数不能正确返回认证状态

### DISCUSSION THREAD

**Discussion by billy-poon** [midwayjs/midway#3609] [General / 闲聊]
[isAuthenticated](https://github.com/midwayjs/midway/blob/55cc29e6754639400e30754c28c69096358ab57a/packages/passport/src/passport/passport.service.ts#L446)

···
     req.isAuthenticated = () => {
        const property = this.passport.getUserProperty();
        return !!this[property];
      };
···

> 此处 `this` 指向 `PassportMiddleware` 的实例，而不是 req

目前我的解决办法是覆写 `attachRequestMethod` 函数

```
...
@Middleware()
export class PassportMiddleware extends CreatePassportMiddleware([LocalStrategy]) {
...
    attachRequestMethod(req: any) {
        // @ts-expect-error
        super.attachRequestMethod(req)

        // 修正 isAuthenticated 不能正确返回认证状态的问题
        req.isAuthenticated = function() {
            return !!this.user
        }
    }
}
```

这样可以解决问题。

不知道是不是我的用法有错？

**Comment by czy88840616** [comment]
感谢反馈，的确是 bug。 #3617
